From bbc1c2a7f06e3a77af70e1e030df047afe6a6d3a Mon Sep 17 00:00:00 2001
From: Luiz Augusto von Dentz <luiz.von.dentz@intel.com>
Date: Tue, 25 Mar 2025 17:30:03 -0400
Subject: [PATCH] main: Fix handling of legacy experimental LL Privacy

Legacy experimental LL Privacy was not enabled by default but
CentralAddressResolution is which may cause problems with old
kernels were LL Privacy may not work as expected.

Fixes: https://github.com/bluez/bluez/issues/1138
---
 src/main.c | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/src/main.c b/src/main.c
index 50346abbb..6a682e9b9 100644
--- a/src/main.c
+++ b/src/main.c
@@ -715,20 +715,11 @@ static bool match_experimental(const void *data, const void *match_data)
 bool btd_kernel_experimental_enabled(const char *uuid)
 {
 	if (!btd_opts.kernel)
-		goto done;
+		return false;
 
 	if (queue_find(btd_opts.kernel, match_experimental, uuid))
 		return true;
 
-done:
-	/* For backward compatibility set LL Privacy as enabled if
-	 * CentralAddressResolution has been set so old kernel LL Privacy is
-	 * enabled.
-	 */
-	if (!strcmp(uuid, "15c0a148-c273-11ea-b3de-0242ac130004") &&
-			btd_opts.defaults.le.addr_resolution)
-		return true;
-
 	return false;
 }
 
